src/main/java/hudson/plugins/git/opt/PreBuildMergeOptions.java

```diff
@@ -50,6 +55,15 @@ public void setMergeTarget(String mergeTarget) {
         this.mergeTarget = mergeTarget;
     }
 
+    @Exported
+    public Integer getTimeout() {
```

Comment:
This needs to be implemented in the `PreBuildMerge` but how do I retrieve the time for which the command has been running.So that we can throw a `TimeoutException`
```
MergeCommand cmd = git.merge().setRevisionToMerge(rev.getSha1());
            for (GitSCMExtension ext : scm.getExtensions())
                ext.decorateMergeCommand(scm, build, git, listener, cmd);
            cmd.execute();
```